docs/details/advisors/checks/Develop Security Checks

```diff
@@ -0,0 +1,260 @@
+2. pmm-managed downloads checks file from SaaS (see Check Service).
+3. pmm-managed verifies file signatures using a list of hard-coded public keys. At least one signature should be correct.
+4. pmm-managed sends queries to pmm-agent and gathers results.
+5. pmm-managed executes check scripts that produce alert information.
+6. pmm-managed sends alerts to Alertmanager.
+    - Due to Alertmanager design, pmm-managed has to send and re-send alerts to it much more often than the frequency of check execution. That's expected, and not important for using STT, but important for understanding how it works.
+    - Currently, Prometheus is not involved.
+## Frontend
+![!](../../../_images/FrontEndSTT.png) 
+Our UI in Grafana uses Alertmanager API v2 to get information about failed security checks.
+
+## Develop custom checks
+
+
+1.  Download the latest PMM Server and PMM Client builds:
+
+    - PMM Server: *docker pull perconalab/pmm-server:dev-latest*
+    - PMM Client: *pmm2-client-2.9.1.tar.gz*
+2. Run PMM Server with special environment variables:
+
+    - *PMM_DEBUG=1* to enable debug output that would be useful later;
+    - *PERCONA_TEST_CHECKS_FILE=/srv/custom-checks.yml* to use checks from the local files instead of downloading them from the SaaS
+    - *PERCONA_TEST_CHECKS_DISABLE_START_DELAY=true* to disable the default check execution start delay, which is currently set to 1 minute, so that checks are run immediately upon system start
+    - *PERCONA_TEST_CHECKS_RESEND_INTERVAL=2s* to define the frequency for sending the SA-based alerts to Alertmanager.
+
+
+
+    ```yaml docker run -p 80:80 -p 443:443 --name pmm-server \
+
+    -e PMM_DEBUG=1 \
+    -e PERCONA_TEST_CHECKS_FILE=/srv/custom-checks.yml \
+    -e PERCONA_TEST_CHECKS_DISABLE_START_DELAY=true
+    -e PERCONA_TEST_CHECKS_RESEND_INTERVAL=2s \
+    perconalab/pmm-server:dev-latest
+
+3. Log in to Grafana (admin/admin) and enable STT in the settings: http://127.0.0.1/graph/d/pmm-settings/pmm-settings
+
+![!](../../../_images/Grafana.png) 
+
+4. Create */srv/custom-checks.yml* inside a Docker container with the content from the Security Advisor (Security Threat Tool) section above.
+
+5. STT checks will run with a time interval defined via UI. You can see the result of running the advisor on the home dashboard:
+
+![!](../../../_images/HomeDashboard.png) 
+
+6. Click on the number of failed checks to open the Failed Checks dashboard:
+
+![!](../../../_images/FailedChecks.png) 
+
+7. Go into Docker container to output the logs of pmm-managed and read STT logs:
+```
+# get inside the container
+docker exec -it pmm-server bash
+# print and watch the logs
+supervisorctl tail -f pmm-managed
+```
+
+
+## Sumbit feedback
+We welcome your feedback on the current process for developing and debugging checks. Send us your comments over [Slack](percona.slack.com) or post a question on the [Percona Forums](https://forums.percona.com/).
+
```

Comment:
@catalinaadam can u please rename the file adding extension .md for a correct file parsing?